### ISSUE TITLE
Columns block: Column width's percentage is NaN when using `,` character as decimal separator

### ISSUE THREAD

**Issue by fluiddot** [WordPress/gutenberg#33601]
## Description

Looks like the percentage field only supports the `.` character as the decimal separator, using other characters like `,` leads to a NaN value. On iOS, depending on the device's region (in my case I used Spain), the keyboard only allows the option to use the `,` character as the decimal separator which prevents inputting decimal numbers (see attached screenshots).

## Step-by-step reproduction instructions

1. Go to device settings and change the device's region to one that has a different decimal separator, like for example Spain. On iOS, this setting can be changed in `Settings/General/Language & Region/Region`.
2. Add a Columns block.
3. Open the block settings.
4. Tap on the width percentage of a column.
5. Input a decimal number using the `,` character separator.
6. Observe that the value is NaN.

## Expected behaviour

The column widths' percentage should support the `,` decimal separator.

## Actual behaviour

The column widths' percentage doesn't support the `,` decimal separator.

## Screenshots or screen recording (optional)

Decimal character|NaN error
--|--
<img src=https://user-images.githubusercontent.com/14905380/126470507-30cc0014-a386-4bba-92f5-3b2135de6297.PNG width=300>|<img src=https://user-images.githubusercontent.com/14905380/126470537-2c629dc2-e359-4251-994d-e966a2212811.PNG width=300>

## WordPress information
- WordPress version: N/A
- Gutenberg version: N/A
- Are all plugins except Gutenberg deactivated? N/A
- Are you using a default theme (e.g. Twenty Twenty-One)? N/A

## Device information
- Device: iPhone 11
- Operating system: iOS 14.2
- WordPress app version: `develop`

**Comment by ceyhun** [comment]
Could be related to https://github.com/WordPress/gutenberg/pull/33434. I had encountered the same with `Button` block and `em` value on that PR, but it was fixed before getting merged.

**Comment by fluiddot** [comment]
> Could be related to #33434. I had encountered the same with `Button` block and `em` value on that PR, but it was fixed before getting merged.

Yeah, it's most likely related although the solution applied there looks didn't cover the case I mentioned. I see that for the Button radius field when I type the `,` character it's removed automatically, but that behavior is not replicated on the column width. In fact, I don't think we should discard the character but support convert decimals with the `,` character.

**Comment by tyxla** [comment]
This is no longer relevant after the React Native implementation was removed in #78747. Closing.
